src/libstd/sys/windows/os.rs

```diff
@@ -89,12 +89,18 @@ impl Iterator for Env {
             let s = slice::from_raw_parts(p, len as usize);
             self.cur = self.cur.offset(len + 1);
 
-            let (k, v) = match s.iter().position(|&b| b == '=' as u16) {
-                Some(n) => (&s[..n], &s[n+1..]),
-                None => (s, &[][..]),
-            };
-            Some((OsStringExt::from_wide(k), OsStringExt::from_wide(v)))
-        }
+            if s.is_empty() {
```

Comment:
This check doesn't need to be here as it was covered above via `if *self.cur == 0`, that's what indicates the end of the environment variable block.